source/parser.py

```diff
@@ -0,0 +1,29 @@
+import os
+import re
+destination = open("../algorithmicbotany.pde", "w")
+pattern = re.compile(r'public\s+class')
+path = "./"
+
+def parse (file, destination):
+  with open(file, 'r') as f:
+    lines = f.readlines()
+    for line in lines:
+      if pattern.match(line):
+        destination.write(re.sub('public', '', line))
+      elif line.find("/*~") >= 0:
```

Comment:
you can do `find` only once and remember the result. that will save the runtime.